data/script.js

```diff
@@ -0,0 +1,60 @@
+const specSelector = document.getElementById("spec");
+const algos = document.getElementById("algos");
+
+const { results } = await (await fetch("./index.json")).json();
+
+specSelector.addEventListener("change", showSpec);
+
+results.sort((a,b) => a.title.localeCompare(b.title)).forEach(s => {
+  if (s.algorithms) {
+    const opt = document.createElement("option");
+    opt.value = s.algorithms;
+    opt.textContent = s.title;
+    specSelector.append(opt);
+  }
+});
+
+function showAlgo(a, level = 0) {
+  const ret = [];
+  if (level === 0) {
+    const heading = document.createElement("h2");
+    if (a.name) {
+      heading.textContent = a.name;
+    } else {
+      heading.textContent = "(unnamed algorithm)";
+    }
+    ret.push(heading);
```

Comment:
It would be great if there could be a link back to the spec when `a.href` is set, either directly from the heading or through a link next to it. Back and forth between extracts and the spec to try to understand what happened, or did not happen, is common :)

Reply:
done in [d25e845](https://github.com/tidoust/parallel-promise/pull/3/commits/d25e84530d374004e12f1c64a4816816b6477f22)